proposed-html-structure-from-wordpress.php

```diff
@@ -0,0 +1,185 @@
+	}
+	function set_id ($post) {
+		$this->id = $post->ID;
+	}
+	function set_taxonomies ($post) {
+		$this->taxonomies = get_the_taxonomies($post);
+	}
+	//only run after set_taxonomies()
+	function set_categories ($post) {
+		if ($this->taxonomies == null) {
+			$this->set_taxonomies($post);
+		}  else {
+			$this->categories = strip_tags($this->taxonomies['category']);
+		}
+	}
+	function set_tags ($post) {
+		if ($this->taxonomies == null) {
+			$this->set_taxonomies($post);
+		} else {
+			$this->tags = strip_tags( $this->taxonomies['post_tag'] );
+		}
+	}
+	function set_guidance ($post) {
+		if ($this->taxonomies == null) {
+			$this->set_taxonomies($post);
+		} else {
+			$this->guidance = strip_tags($this->taxonomies['guidance']);
+		}
+	}
+	//can only run after set_guidance()
+	function set_recommended () {
+		if (!strpos($this->guidance, 'Recommended')) {
+			$this->recommended = 'false';
+		} else {
+			$this->recommended = 'true';
+		}
+	}
+	function set_online () {
+			if (!strpos($this->guidance, 'All online')) {
+			$this->online = 'false';
+		} else {
+			$this->online = 'true';
+		}
+	}
+	//can only run after set_categories()
+	function set_partners () {
+		if (strpos($this->categories, 'Findmypast.co.uk') !== false || strpos($this->categories, 'Ancestry.co.uk') !== false ) {
+			$this->partners = 'true';
+		} else {
+			$this->partners = 'false';
+		}
+	}
+
+	function get_url () {return $this->url;}
+	function get_title () {return $this->title;}
+	function get_slug () {return $this->slug;}
+	function get_id () {return $this->id;}
+	function get_taxonomies () {return $this->taxonomies;}
+	function get_categories () {return $this->categories;}
+	function get_tags () {return $this->tags;}
+	function get_guidance () {return $this->guidance;}
+	function get_recommended () {return $this->recommended;}
+	function get_online () {return $this->online;}
+	function get_partners () {return $this->partners;}
+
+	function print_html_structure () {
+		$html = '<div data-name="%s"
+					     data-categories="%s"
+					     data-recommended-guide-for-category="%s"
+					     data-keywords="%s"
+					     data-all-records-available-online="%s"
+					     data-guide-href="%s"
+					     data-available-on-partner="%s"
+					     id="%s"
+					     class="research-guide">
+					    <h3>
+					        <a href="%s">
+					            %s
+					        </a>
+					    </h3>
+					</div>
+					';
+
+		$compiled = sprintf( $html, $this->title, $this->guidance, $this->recommended, $this->tags, $this->online, $this->url, $this->partners, $this->slug, $this->url, $this->title);
+
+		echo $compiled;
+	}
+}
+
+
+
+function get_research_guide_array ($category) {
+	$args = array (
+		'category_name' => $category,
+		'post_type' => 'page',
+		'posts_per_page' => -1,
+		'post_status' => 'publish'
+	);
+	$research_guide_query = new WP_Query($args);
+	$research_guide_array = array();
+
+	// The Loop
+	if ( $research_guide_query->have_posts() ) :
```

Comment:
This alternative syntax for control structures is best used when you've got PHP interspersed within HTML (because it makes it clearer where the control structure ends, rather than a closing `}`). In this case I think you should use the normal syntax.